src/RoslynInsertionTool/RoslynInsertionTool/RoslynInsertionTool.VisualStudioTeamServices.cs

```diff
@@ -552,27 +554,69 @@ private static async Task<string> GetComponentBuildDropLogAsync(Build newestBuil
             {
+                return await GetChangesBetweenBuildsFromGitHubAsync(repoId, fromSHA, toSHA);
+            }
+            else if (tobuild.Repository.Type == "TfsGit")
+            {
+                return await GetChangesBetweenBuildsFromAzDOAsync(tobuild, repoId, fromSHA, toSHA);
+            }
 
-                var restEndpoint = $"https://api.github.com/repos/{repoId}/compare/{fromSHA}...{toSHA}";
-                var client = new HttpClient();
-                var request = new HttpRequestMessage(HttpMethod.Get, restEndpoint);
-                request.Headers.Add("User-Agent", "RoslynInsertionTool");
-
-                var response = await client.SendAsync(request);
-                var content = await response.Content.ReadAsStringAsync();
+            throw new NotSupportedException("Only builds created from GitHub & AzDO repos support enumerating commits.");
+        }
 
-                // https://developer.github.com/v3/repos/commits/
-                var data = JsonConvert.DeserializeAnonymousType(content, new
+        private static async Task<(List<GitCommit> changes, string diffLink)> GetChangesBetweenBuildsFromAzDOAsync(Build tobuild, string repoId, string fromSHA, string toSHA)
+        {
+            var gitClient = ComponentBuildConnection.GetClient<GitHttpClient>();
+            var getCommits = (await gitClient.GetCommitsAsync(
+                repoId,
+                new GitQueryCommitsCriteria()
                 {
-                    commits = new[]
+                    ItemVersion = new GitVersionDescriptor() { Version = fromSHA, VersionType = GitVersionType.Commit },
+                    CompareVersion = new GitVersionDescriptor() { Version = toSHA, VersionType = GitVersionType.Commit }
+                }))
+                // AzDO does not provide the full commit message, so we must query for each commit to provide better messages for PR merge commits.
+                .Select(c => gitClient.GetCommitAsync(c.CommitId, repoId));
+            var commits = (await Task.WhenAll(getCommits))
+                .Select(c =>
+                    new GitCommit()
                     {
+                        Author = c.Author.Name,
+                        Committer = c.Committer.Name,
+                        CommitDate = c.Committer.Date,
+                        Message = c.Comment,
+                        CommitId = c.CommitId,
+                        RemoteUrl = c.Links.Links["web"].ToString()
+                    })
+                .ToList();
+
+            // AzDO does not have a UI for comparing two commits. Instead generate the REST API call to retrieve commits between two SHAs.
+            return (commits, $"{tobuild.Repository.Url.OriginalString.Replace("_git", "_apis/git/repositories")}/commits?searchCriteria.itemVersion.version={fromSHA}&searchCriteria.itemVersion.versionType=commit&searchCriteria.compareVersion.version={toSHA}&searchCriteria.compareVersion.versionType=commit");
```

Comment:
once we merge, it might be good to ask for feedback from the testing team, and maybe they can jump in to implement any refinements to the behavior that they want.